punishments.js

```diff
@@ -28,6 +30,30 @@ let nameLockedUsers = Punishments.nameLockedUsers = Object.create(null);
 let lockedRanges = Punishments.lockedRanges = Object.create(null);
 let rangelockedUsers = Punishments.rangeLockedUsers = Object.create(null);
 
+// load ipbans at our leisure
+let loadBanlist = Punishments.loadBanlist = function () {
+	return new Promise(function (resolve, reject) {
+		fs.readFile(path.resolve(__dirname, 'config/ipbans.txt'), (err, data) => {
+			if (err) return reject();
+			data = ('' + data).split("\n");
+			let rangebans = [];
+			for (let i = 0; i < data.length; i++) {
+				data[i] = data[i].split('#')[0].trim();
+				if (!data[i]) continue;
+				if (data[i].includes('/')) {
+					rangebans.push(data[i]);
+				} else if (!bannedIps[data[i]]) {
+					bannedIps[data[i]] = '#ipban';
+				}
+			}
+			Punishments.checkRangeBanned = Cidr.checker(rangebans);
+			resolve();
+		});
+	});
+};
+
+loadBanlist();
```

Comment:
Yes, the original code was at our leisure. You moved the code to a point where it wasn't done at our leisure at all.